e2e/driver/CheReporter.ts

```diff
@@ -45,10 +49,44 @@ class CheReporter extends mocha.reporters.Spec {
       TS_SELENIUM_HAPPY_PATH_WORKSPACE_NAME: ${TestConstants.TS_SELENIUM_HAPPY_PATH_WORKSPACE_NAME}
       TS_SELENIUM_USERNAME: ${TestConstants.TS_SELENIUM_USERNAME}
       TS_SELENIUM_PASSWORD: ${TestConstants.TS_SELENIUM_PASSWORD}
+      TS_SELENIUM_DELAY_BETWEEN_SCREENSHOTS: ${TestConstants.TS_SELENIUM_DELAY_BETWEEN_SCREENSHOTS}
+      TS_SELENIUM_REPORT_FOLDER: ${TestConstants.TS_SELENIUM_REPORT_FOLDER}
+      TS_SELENIUM_EXECUTION_SCREENCAST: ${TestConstants.TS_SELENIUM_EXECUTION_SCREENCAST}
+      DELETE_SCREENCAST_IF_TEST_PASS: ${TestConstants.DELETE_SCREENCAST_IF_TEST_PASS}
 
 ########################################################
       `;
       console.log(launchInformation);
+
+      rm.sync(TestConstants.TS_SELENIUM_REPORT_FOLDER);
+    });
+
+    runner.on('test', async function (test: mocha.Test) {
+      if (!TestConstants.TS_SELENIUM_EXECUTION_SCREENCAST) {
+        return;
+      }
+
+      methodIndex = methodIndex + 1;
+      const currentMethodIndex: number = methodIndex;
+      let iterationIndex: number = 1;
+
+      while (!(test.state === 'passed' || test.state === 'failed')) {
+        await screenCatcher.catchMethodScreen(test.title, currentMethodIndex, iterationIndex);
+        iterationIndex = iterationIndex + 1;
+
+        await driverHelper.wait(TestConstants.TS_SELENIUM_DELAY_BETWEEN_SCREENSHOTS);
+      }
+    });
+
+    runner.on('test end', async function (test: mocha.Test) {
+      if (!TestConstants.TS_SELENIUM_EXECUTION_SCREENCAST) {
+        return;
+      }
+
+      const currentMethodIndex: number = methodIndex;
+      let iterationIndex: number = 10000;
```

Comment:
It seems quite weird to set the `iterationIndex` to some random number. What about having `iterationIndex` as a global variable? Then you can simply set `iterationIndex + 1` here and set `iterationIndex` to 1 after that.

Reply:
It is just a way to mark the last screenshot.

Reply:
Yeah, I understand what you are doing, but I think it can be done better. Why to have screens indexed [1. 2. 10000] and not [1. 2. 3].

Reply:
Because it is too hard to find the last screenshot if use your suggested approach

![Screenshot from 2019-08-01 14-41-30](https://user-images.githubusercontent.com/22724619/62290678-93c1de80-b46a-11e9-97eb-ff81c2930c91.png)